### PR TITLE
Added docstrings code showing example of using `metrics` with `evaluate`

### PR DESCRIPTION
**Pull request by TNTran92**
#### Reference Issues/PRs

Issue #2849

#### What does this implement/fix? Explain your changes.

This PR adds docstrings code showing how to use the `scoring` param with `metrics` in order to evaluate both point forecast and interval forecast.

#### Does your contribution introduce a new dependency? If yes, which one?

No
##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [x] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [x] I've added unit tests and made sure they pass locally.
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.

##### For new estimators
- [x] I've added the estimator to the online documentation.
- [x] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Review by aiwalter — COMMENTED** [review]
added some comments, can you also pls mention in 1 or 2 sentences that when you want to evaluate prediction intervals, that givin a loss that can do so will automatically make the forecaster to predict the intervals?

File: sktime/forecasting/model_evaluation/_functions.py [hunk 1]
```diff
@@ -34,45 +34,57 @@ def evaluate(
...
+          only fitted on the first train window data and then updated,
+          or always refitted.
+      scoring : subclass of sktime.performance_metrics.BaseMetric, default=None.
+          Used to get a score function that takes y_pred and y_test arguments
+          and accept y_train as keyword argument.
+          If None, then uses scoring = MeanAbsolutePercentageError(symmetric=True).
+      return_data : bool, default=False
+          Returns three additional columns in the DataFrame, by default False.
+          The cells of the columns contain each a pd.Series for y_train,
+          y_pred, y_test.
```
[the remark this answers is not in the document]
Comment (TNTran92) [reply sktime/forecasting/model_evaluation/_functions.py]: I have no idea how tthey end up there. I have removed them

File: sktime/forecasting/model_evaluation/_functions.py [hunk 2]
```diff
@@ -34,45 +34,57 @@ def evaluate(
...
 
     Returns
     -------
-    pd.DataFrame
-        DataFrame that contains several columns with information regarding each
-        refit/update and prediction of the forecaster.
+      pd.DataFrame
+          DataFrame that contains several columns with information regarding each
+          refit/update and prediction of the forecaster.
```
[the remark this answers is not in the document]
Comment (TNTran92) [reply sktime/forecasting/model_evaluation/_functions.py]: They've been removed

File: sktime/forecasting/model_evaluation/_functions.py [hunk 3]
```diff
@@ -34,45 +34,57 @@ def evaluate(
...
+      >>> from sktime.forecasting.model_selection import ExpandingWindowSplitter
+      >>> from sktime.forecasting.naive import NaiveForecaster
+      >>> y = load_airline()
+      >>> forecaster = NaiveForecaster(strategy="mean", sp=12)
+      >>> cv = ExpandingWindowSplitter(
+        initial_window=24,
+        step_length=12,
+        fh=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12])
+      >>> results = evaluate(forecaster=forecaster, y=y, cv=cv)
+    -------
```
[the remark this answers is not in the document]
Comment (TNTran92) [reply sktime/forecasting/model_evaluation/_functions.py]: Added some sections. A new commit is up.

File: sktime/forecasting/model_evaluation/_functions.py [hunk 4]
```diff
@@ -62,17 +63,44 @@ def evaluate(
 
     Examples
     --------
+        The type of evaluation that is done by `evaluate` depends on metrics in
+        param `scoring`
+        When evaluating model/estimators on point forecast, users can either let
```
Comment (fkiraly) [inline sktime/forecasting/model_evaluation/_functions.py:68]: remove "either"; since there is not an "or"

File: sktime/forecasting/model_evaluation/_functions.py [hunk 5]
```diff
@@ -44,7 +44,8 @@ def evaluate(
         Exogenous variables
     strategy : {"refit", "update"}
         Must be "refit" or "update". The strategy defines whether the `forecaster` is
-        only fitted on the first train window data and then updated, or always refitted.
+        only fitted on the first train window data and then updated,
```
Comment (fkiraly) [inline sktime/forecasting/model_evaluation/_functions.py:47]: why did you shorten this?

File: sktime/forecasting/model_evaluation/_functions.py [hunk 6]
```diff
@@ -62,17 +63,44 @@ def evaluate(
...
     >>> y = load_airline()
     >>> forecaster = NaiveForecaster(strategy="mean", sp=12)
-    >>> cv = ExpandingWindowSplitter(
-    ...     initial_window=24,
-    ...     step_length=12,
-    ...     fh=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12])
+    >>> cv = ExpandingWindowSplitter(initial_window=12, step_length=3, fh=[1, 2, 3])
     >>> results = evaluate(forecaster=forecaster, y=y, cv=cv)
+
+        Optionally, users may select other metrics that are compatible
```
Comment (fkiraly) [inline sktime/forecasting/model_evaluation/_functions.py:79]: replace "comaptible with point for cast" better with "metrics can be supplied by the `scoring` argument. These can be forecast metrics of any kind, i.e., point forecast metrics, interval metrics, quantile foreast metrics.

File: sktime/forecasting/model_evaluation/_functions.py (hunk 5 above)
Comment (TNTran92) [reply sktime/forecasting/model_evaluation/_functions.py:47]: THe rest of the sentence is on a new line. I must've moved it becuase of fear the sentence would exceed max line number. I have moved it back.

File: sktime/forecasting/model_evaluation/_functions.py (hunk 4 above)
Comment (TNTran92) [reply sktime/forecasting/model_evaluation/_functions.py:68]: Done

File: sktime/forecasting/model_evaluation/_functions.py [hunk 7]
```diff
@@ -62,17 +63,44 @@ def evaluate(
...
     >>> from sktime.forecasting.model_evaluation import evaluate
     >>> from sktime.forecasting.model_selection import ExpandingWindowSplitter
     >>> from sktime.forecasting.naive import NaiveForecaster
     >>> y = load_airline()
     >>> forecaster = NaiveForecaster(strategy="mean", sp=12)
-    >>> cv = ExpandingWindowSplitter(
-    ...     initial_window=24,
-    ...     step_length=12,
-    ...     fh=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12])
+    >>> cv = ExpandingWindowSplitter(initial_window=12, step_length=3, fh=[1, 2, 3])
```
[the remark this answers is not in the document]
Comment (TNTran92) [reply sktime/forecasting/model_evaluation/_functions.py:76]: I did that to make the line shorter. But I've added it back.

File: sktime/forecasting/model_evaluation/_functions.py (hunk 6 above)
Comment (TNTran92) [reply sktime/forecasting/model_evaluation/_functions.py:79]: Done

File: sktime/forecasting/model_evaluation/_functions.py (hunk 5 above)
Comment (fkiraly) [reply sktime/forecasting/model_evaluation/_functions.py:47]: Thanks! It does not look like you have moved it back though?

Max line length is 88 characters for `sktime`, see `setup.cfg`.
If you are using vscode and have linting turned on, it should automatically watch out for that.

You can also add rulers to your editor windows:
https://stackoverflow.com/questions/29968499/vertical-rulers-in-visual-studio-code

**Comment by fkiraly** [comment]
Thanks! I think you haven´t committed your changes yet, though?

**Comment by TNTran92** [comment]
> Thanks! I think you haven´t committed your changes yet, though?

I'm still running pytest locally. There is a weird module that's causing it to fail

File: sktime/forecasting/model_evaluation/_functions.py (hunk 5 above)
Comment (TNTran92) [reply sktime/forecasting/model_evaluation/_functions.py]: On line 47 of commit [38c3351](https://github.com/alan-turing-institute/sktime/pull/2850/commits/38c3351f39ed14b254fb1a624e8d79993b521d51), i think it's back. This commit is included in [1d6361b](https://github.com/alan-turing-institute/sktime/pull/2850/commits/1d6361b727f9ec12a36e66a6bf73adab19dc60eb), which is being tested. Do you see the full sentence there?

File: sktime/forecasting/model_evaluation/_functions.py [hunk 8]
```diff
@@ -62,17 +62,47 @@ def evaluate(
...
+    ... fh=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12])
     >>> results = evaluate(forecaster=forecaster, y=y, cv=cv)
+
+        Optionally, users may select other metrics that can be supplied
+        by `scoring` argument.
+        These can be forecast metrics of any kind, i.e., point forecast metrics,
+        interval metrics, quantile foreast metrics.
+        https://www.sktime.org/en/stable/api_reference/performance_metrics.html?highlight=metrics
+
+        To evaluate models/estimators using point forecast with these metrics,
```
Comment (fkiraly) [inline sktime/forecasting/model_evaluation/_functions.py:85]:
```suggestion
        To evaluate models/estimators using a specific metric, provide them to the scoring arg
```

File: sktime/forecasting/model_evaluation/_functions.py [hunk 9]
```diff
@@ -62,17 +62,47 @@ def evaluate(
...
+        simply add that metrics as follows
+    >>> from sktime.performance_metrics.forecasting import MeanAbsoluteError
+    >>> loss = MeanAbsoluteError()
+    >>> results = evaluate(forecaster=forecaster, y=y, cv=cv, scoring=loss)
+
+        When evaluating models/estimators on interval
```
Comment (fkiraly) [inline sktime/forecasting/model_evaluation/_functions.py:91]:
```suggestion
        An example of an interval metric is the PinballLoss.
        It can be used with all probabilistic forecasters, which can be listed as follows:
```

File: sktime/forecasting/model_evaluation/_functions.py [hunk 10]
```diff
@@ -62,17 +62,47 @@ def evaluate(
...
+        predictions or quantile predictions, the PinballLoss
+        metrics can be used. The list of estimators that are compatible with
+        this metrics are
+    >>> from sktime.registry import all_estimators
+    >>> all_estimators("forecaster", filter_tags={"capability:pred_int": True},
+    ... as_dataframe=True)
+
+        Evaluate estimators using PinballLoss
```
Comment (fkiraly) [inline sktime/forecasting/model_evaluation/_functions.py:99]:
```suggestion
        To evaluate probabilistic forecasts, simply pass a probabilistic metric like PinballLoss:
```

**Comment by fkiraly** [comment]
made some suggestions to improve the formulations, otherwise great and ready to go

**Comment by fkiraly** [comment]
@TNTran92, you've somehow kept the second and third line where what I wrote was meant to replace entire paragraphs. Can you kindly check? Lots of sentences that start halfway now...

**Comment by TNTran92** [comment]
> @TNTran92, you've somehow kept the second and third line where what I wrote was meant to replace entire paragraphs. Can you kindly check? Lots of sentences that start halfway now...

Right...at some point, I don't even understand what was written anymore... I fixed it. Does it make sense now?

**Comment by fkiraly** [comment]
I'd remove "which can be used as follows:", otherwise looks good!

**Comment by TNTran92** [comment]
> I'd remove "which can be used as follows:", otherwise looks good!

Removed and committed!

**Comment by TNTran92** [comment]
All checks have passed
